### PR TITLE
Some changes

### PR DESCRIPTION
**Pull request by KCodeYT**
This pull request includes:
- appending `this` on instance based variables
- ProxyServer#getServers() now returns an unmodicable collection instead of a new hashset, improves speed and makes it immutable.
- Creating a static final instance of the WaterdogScheduler and make its constructor private so no one can access it.
- Also validating the client signed tokens.

### CODE REVIEW

**Comment by TobiasG-DE** [comment]
Will Review

File: src/main/java/dev/waterdog/player/HandshakeUtils.java [hunk 1]
```diff
@@ -72,7 +72,9 @@ public static boolean validateChain(JsonArray chainArray) throws Exception {
             }
 
             if (lastKey != null) {
-                EncryptionUtils.verifyJwt(jwt, lastKey);
+                if (!EncryptionUtils.verifyJwt(jwt, lastKey)) {
+                    return false;
```
Comment (TobiasG-DE) [inline src/main/java/dev/waterdog/player/HandshakeUtils.java:76]: If the encryption verification fails this is supposed to throw an exception which we can use to detect whether it was successful or not ( see [here](https://github.com/CloudburstMC/Protocol/blob/8ec21ae916ed6b1338246920aa1657dcb4099339/bedrock/bedrock-common/src/main/java/com/nukkitx/protocol/bedrock/util/EncryptionUtils.java#L105) )

**Review by TobiasG-DE — COMMENTED** [review]
Overall a good refactor. I believe we should have a look on how to treat encryption verification, everything else looks good

File: src/main/java/dev/waterdog/player/HandshakeUtils.java (hunk 1 above)
Comment (KCodeYT) [reply src/main/java/dev/waterdog/player/HandshakeUtils.java:76]: But false already says that the decryption did failed, but I can make it throw an exception

Comment (TobiasG-DE) [reply src/main/java/dev/waterdog/player/HandshakeUtils.java:76]: Hmm I'm wondering why it has both a return value and can throw an exception. Probably returns false if the validation fails and throws an exception If the key is unusable or smth, I'll look into that later

Comment (KCodeYT) [reply src/main/java/dev/waterdog/player/HandshakeUtils.java:76]: So should I remove this part or can you accept it like this?

Comment (TobiasG-DE) [reply src/main/java/dev/waterdog/player/HandshakeUtils.java:76]: I guess I can approve here, returning false and an exception should both work.

**Review by TobiasG-DE — APPROVED** [review]
Waiting for @Alemiz112 review

File: src/main/java/dev/waterdog/network/rewrite/types/BlockPalette.java [hunk 2]
```diff
@@ -32,7 +32,7 @@
  */
 public class BlockPalette {
 
-    private static final Int2ObjectMap<BlockPalette> paletteCache = new Int2ObjectOpenHashMap<>();
+    private static final Int2ObjectMap<BlockPalette> CACHE = new Int2ObjectOpenHashMap<>();
```
Comment (Alemiz112) [inline src/main/java/dev/waterdog/network/rewrite/types/BlockPalette.java:35]: I don't see the point of changing this. Also please leave it lowercase based.

File: src/main/java/dev/waterdog/player/HandshakeUtils.java [hunk 3]
```diff
@@ -45,20 +45,20 @@
  */
 public class HandshakeUtils {
 
-    private static final KeyPair privateKeyPair;
+    private static final KeyPair PRIVATE_KEY_PAIR;
```
Comment (Alemiz112) [inline src/main/java/dev/waterdog/player/HandshakeUtils.java:48]: Same here.

File: src/main/java/dev/waterdog/player/HandshakeUtils.java (hunk 1 above)
Comment (Alemiz112) [reply src/main/java/dev/waterdog/player/HandshakeUtils.java:76]: Returning false is correct.

File: src/main/java/dev/waterdog/scheduler/WaterdogScheduler.java [hunk 4]
```diff
@@ -26,7 +26,7 @@
 
 public class WaterdogScheduler {
 
-    private static WaterdogScheduler instance;
+    private static final WaterdogScheduler INSTANCE = new WaterdogScheduler(ProxyServer.getInstance());
```
Comment (Alemiz112) [inline src/main/java/dev/waterdog/scheduler/WaterdogScheduler.java:29]: At this point `ProxyServer.getInstance()` will return null. Also don't forget about lowercase😉

**Review by Alemiz112 — CHANGES_REQUESTED** [review]

File: src/main/java/dev/waterdog/scheduler/WaterdogScheduler.java (hunk 4 above)
Comment (KCodeYT) [reply src/main/java/dev/waterdog/scheduler/WaterdogScheduler.java:29]: The proxy instance gets initiated before the waterdog scheduler so it cant be null

Comment (Alemiz112) [reply src/main/java/dev/waterdog/scheduler/WaterdogScheduler.java:29]: Scheduler instance is initialized statically so it is initialized before proxy starts.

Comment (KCodeYT) [reply src/main/java/dev/waterdog/scheduler/WaterdogScheduler.java:29]: No the static method gets called when the class loader loads it and this happens when a static method gets executed, an instance gets created or the class gets loaded with `Class.forName()`

Comment (KCodeYT) [reply src/main/java/dev/waterdog/scheduler/WaterdogScheduler.java]: I could create a test so the build would fail if the instance is null

Comment (Alemiz112) [reply src/main/java/dev/waterdog/scheduler/WaterdogScheduler.java]: Alright. I see what you mean now. Also I'm thinking about allowing plugins to register create own scheduler which would extend `WaterdogScheduler` class so instance would be created after plugins are loaded.

**Review by Alemiz112 — APPROVED** [review]
Aproved.

**Pull request merged**